M71 MAESTRÍA EN GESTIÓN Y ANÁLISIS DE DATOS FINANCIEROS

# METODOLOGIAS AVANZADAS DE APRENDIZAJE AUTOMATICO

Macarena Benjamin


**TP 2**

# **Extraccion de los datos**

Para el analisis se elige la accion BUMBLE que es una aplicación de redes y citas en línea. Su salida a bolsa se hizo bajo el nombre Bumble Inc.(BMBL) en febrero de 2021.

In [ ]:
!pip install mplfinance
import pandas as pd
import yfinance as yf
import mplfinance as mpf
from datetime import datetime

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 75.0/75.0 kB 3.1 MB/s eta 0:00:00


In [ ]:
#Descargar datos de Bumble desde Yahoo Finance
ticker = 'BMBL'
datos = yf.download(ticker, end='2024-10-01')
datos

[*********************100%***********************]  1 of 1 completed


Open       High        Low      Close  Adj Close    Volume
Date                                                                       
2021-02-11  76.000000  79.599998  70.000000  70.309998  70.309998  43502700
2021-02-12  75.709999  84.800003  71.510002  75.459999  75.459999  21414200
2021-02-16  79.150002  79.599998  70.377998  78.889999  78.889999  13163000
2021-02-17  76.779999  78.697998  75.000000  77.320000  77.320000   3769600
2021-02-18  75.449997  76.400002  73.629997  74.000000  74.000000   3166400
...               ...        ...        ...        ...        ...       ...
2024-09-24   5.930000   6.295000   5.930000   6.240000   6.240000   2623400
2024-09-25   6.220000   6.490000   6.190000   6.320000   6.320000   2231700
2024-09-26   6.496000   6.580000   6.210000   6.440000   6.440000   2609900
2024-09-27   6.350000   6.650000   6.310000   6.430000   6.430000   2733100
2024-09-30   6.420000   6.650000   6.350000   6.380000   6.380000   1764500

[914 rows x 6 columns]

In [ ]:
#Ver cuantos registros hay
print(f"Numero de registros: {len(datos)}")

Numero de registros: 914


Se verifica que comienza su actividad el 11 de febrero del 2021 y se hace un corte en el analisis al finalizar el mes de septiembre 2024

# **Generacion de imagenes y Clasificacion**

In [ ]:
import os
import pandas as pd
import mplfinance as mpf
from PIL import Image

#Funcion para generar los candlestick chart, recortar y ajustar la imagen
def generar_imagen_candlestick(datos, archivo):
    mpf.plot(
        datos,
        type='candle',
        style=mpf.make_mpf_style(gridstyle='', facecolor='white'),
        title='',
        ylabel='',
        volume=True,
        ylabel_lower='',
        show_nontrading=True,
        savefig=archivo
    )

    #Abrir, recortar y ajustar la imagen
    im = Image.open(archivo)
    im2 = im.crop((145, 70, 720, 473))
    im3 = im2.resize((400, 400))
    im3.save(archivo)

#Funcion para clasificar y guardar las imágenes en carpetas
def clasificar_y_guardar(precio_cierre_anterior, precio_cierre_actual, archivo):

     #Calcular el porcentaje de variacion de precio basado en el precio actual
    cambio_porcentual = (precio_cierre_actual - precio_cierre_anterior) / precio_cierre_actual * 100


    #Clasificacion
    if cambio_porcentual > 1: #Sugerencia de Compra
        carpeta = "Comprar"
    elif cambio_porcentual < -1: #Sugerencia de Venta
        carpeta = "Vender"
    else:  #Sugerencia de Mantener
        carpeta = "Hold"

    #Crear la carpeta si no existe
    if not os.path.exists(carpeta):
        os.makedirs(carpeta)

    #Mover la imagen a la carpeta correspondiente
    nuevo_archivo = os.path.join(carpeta, os.path.basename(archivo))
    os.rename(archivo, nuevo_archivo)
    print(f"Image moved to: {nuevo_archivo}")

#Loop para generar y clasificar las imagenes para cada dia habil
for fecha_actual in datos.index:
    #Extraer los ultimos 5 datos disponibles antes de la fecha actual (no incluir la fecha actual)
    ultimos_datos = datos.loc[:fecha_actual].iloc[:-1].tail(5)

    #Verificar que hay al menos 5 dias de datos disponibles
    if len(ultimos_datos) < 5:
        print(f"No hay suficientes datos para generar la imagen para {fecha_actual}.")
        continue  #Si no hay suficientes datos, saltar a la siguiente fecha


    #Extraer el precio de cierre del dia actual
    precio_cierre_actual = datos.loc[fecha_actual, 'Close']  #Precio de cierre del dia actual

    #Extraer el precio de cierre del ultimo dia de los 5 dias anteriores
    precio_cierre_anterior = ultimos_datos['Close'].iloc[-1]  #Ultimo dia de los 5 seleccionados

    #Definir el nombre de la imagen
    nombre_archivo = f'BMBL_{fecha_actual.strftime("%Y-%m-%d")}.png'

    #Generar la imagen del candlestick
    generar_imagen_candlestick(ultimos_datos, nombre_archivo)

    #Clasificar y guardar en la carpeta
    clasificar_y_guardar(precio_cierre_anterior, precio_cierre_actual, nombre_archivo)


No hay suficientes datos para generar la imagen para 2021-02-11 00:00:00.
No hay suficientes datos para generar la imagen para 2021-02-12 00:00:00.
No hay suficientes datos para generar la imagen para 2021-02-16 00:00:00.
No hay suficientes datos para generar la imagen para 2021-02-17 00:00:00.
No hay suficientes datos para generar la imagen para 2021-02-18 00:00:00.
Image moved to: Vender/BMBL_2021-02-19.png
Image moved to: Vender/BMBL_2021-02-22.png
Image moved to: Vender/BMBL_2021-02-23.png
Image moved to: Comprar/BMBL_2021-02-24.png
Image moved to: Vender/BMBL_2021-02-25.png
Image moved to: Comprar/BMBL_2021-02-26.png
Image moved to: Comprar/BMBL_2021-03-01.png
Image moved to: Vender/BMBL_2021-03-02.png
Image moved to: Vender/BMBL_2021-03-03.png
Image moved to: Vender/BMBL_2021-03-04.png
Image moved to: Comprar/BMBL_2021-03-05.png
Image moved to: Vender/BMBL_2021-03-08.png
Image moved to: Comprar/BMBL_2021-03-09.png
Image moved to: Hold/BMBL_2021-03-10.png
Image moved to: Comprar/B

# **Almacenamiento en Google Drive**

In [ ]:
import shutil
import os

#Abrir Google Drive
from google.colab import drive
drive.mount('/content/drive')

#Crear la carpeta base en Google Drive si no existe
carpeta_destino_base = '/content/drive/My Drive/BMBL_Clasificadas/'
if not os.path.exists(carpeta_destino_base):
    os.makedirs(carpeta_destino_base)

#Definir las rutas locales de las carpetas clasificadas
carpeta_local_buy = './Comprar'
carpeta_local_sell = './Vender'
carpeta_local_hold = './Hold'

#Mover las carpetas "Comprar", "Vender" y "Hold" a Google Drive
shutil.move(carpeta_local_buy, os.path.join(carpeta_destino_base, 'Comprar'))
shutil.move(carpeta_local_sell, os.path.join(carpeta_destino_base, 'Vender'))
shutil.move(carpeta_local_hold, os.path.join(carpeta_destino_base, 'Hold'))




Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


'/content/drive/My Drive/BMBL_Clasificadas/Hold'

In [ ]:
import os

#Definir la ruta base en Google Drive
carpeta_destino_base = '/content/drive/My Drive/BMBL_Clasificadas/'

#Definir las rutas de las carpetas para cada grupo
carpeta_comprar = os.path.join(carpeta_destino_base, 'Comprar')
carpeta_vender = os.path.join(carpeta_destino_base, 'Vender')
carpeta_hold = os.path.join(carpeta_destino_base, 'Hold')

#Funcion para contar imagenes en una carpeta
def contar_imagenes(carpeta):
    if os.path.exists(carpeta):
        return len([archivo for archivo in os.listdir(carpeta) if archivo.endswith('.png')])
    return 0

#Contar las imagenes en cada carpeta
imagenes_comprar = contar_imagenes(carpeta_comprar)
imagenes_vender = contar_imagenes(carpeta_vender)
imagenes_hold = contar_imagenes(carpeta_hold)

#Mostrar los resultados
print(f"Imágenes en 'Comprar': {imagenes_comprar}")
print(f"Imágenes en 'Vender': {imagenes_vender}")
print(f"Imágenes en 'Hold': {imagenes_hold}")


Imágenes en 'Comprar': 298
Imágenes en 'Vender': 376
Imágenes en 'Hold': 235


Se verifica que de los 914 registros que hay para esta accion (con corte en ultimo dia habil de septiembre 2024), se crearon 909 imagenes.

Las imagenes se crean con la data de los ultimos 5 registros (sin incluir el dia que en el que se toma esa imagen).

Los dias que no hay imagenes son los primeros 5 dias dado que no cuentan con registros suficientes para generarlas.


La salida a bolsa fue el jueves 11 febrero, luego le sigue el 12 y ya despues 13 y 14 son sabado y domingo que son dias que la bolsa no opera. El lunes 15 es feriado y luego 16, 17 y 18.
De esta manera se verifica que el primer dia habil que cuenta con 5 registros previos para generar una imagen es el 19 de febrero.

Si a 914 le restamos esos 5 dias se obtiene un total de 909 imagenes.

In [ ]:
image_count = sum([len(files) for r, d, files in os.walk(carpeta_destino_base)])
print(f"Número total de imágenes en la carpeta: {image_count}")

Número total de imágenes en la carpeta: 909


#**Generador de entrenamiento y validacion**

In [ ]:
import os
import tensorflow as tf
from google.colab import drive

#Abrir Google Drive
drive.mount('/content/drive')

#Tamaño de lote
batch_size = 128

#Ruta a la carpeta de imagenes en Google Drive
carpeta_destino_base = '/content/drive/My Drive/BMBL_Clasificadas/'

#Crear un ImageDataGenerator para preprocesar las imagenes
train_datagen = tf.keras.preprocessing.image.ImageDataGenerator(
    zoom_range=1.0,
    validation_split=0.2  #Se agrega el porcentaje de validacion
)

#Crear un generador de entrenamiento desde las carpetas
train_generator = train_datagen.flow_from_directory(
    carpeta_destino_base,  #Directorio donde estan las imagenes
    target_size=(100, 100),  #Redimensionar las imagenes a 100x100
    #color_mode='grayscale',
    batch_size=batch_size,
    classes=['Comprar', 'Vender', 'Hold'],  #Clases
    subset='training'  #Entrenamiento
)

#Crear un generador de validacion
validation_generator = train_datagen.flow_from_directory(
    carpeta_destino_base,
    target_size=(100, 100),
    #color_mode='grayscale',
    batch_size=batch_size,
    classes=['Comprar', 'Vender', 'Hold'],
    subset='validation'  #Validacion
)


Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
Found 728 images belonging to 3 classes.
Found 181 images belonging to 3 classes.


# **Modelo**

In [ ]:
model = tf.keras.models.Sequential([

    # Entrada
    tf.keras.layers.Input(shape=(100, 100, 3)), # 3 en color, 1 es grayscale

    # Convolución 1
    tf.keras.layers.Conv2D(16, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 2
    tf.keras.layers.Conv2D(32, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 3
    tf.keras.layers.Conv2D(64, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 4
    tf.keras.layers.Conv2D(64, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 5
    tf.keras.layers.Conv2D(64, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Achatamiento
    tf.keras.layers.Flatten(),

    # Capas densas
    tf.keras.layers.Dense(100, activation='relu'),

    # cantidad de neuronas de salida, una por clase (como son 3 clases, se coloca 3)
    tf.keras.layers.Dense(3, activation='softmax')

])

model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])

#model.summary()

n_epochs = 10
history = model.fit(train_generator, epochs=n_epochs, verbose=2, validation_data=validation_generator )

Epoch 1/10


/usr/local/lib/python3.10/dist-packages/keras/src/trainers/data_adapters/py_dataset_adapter.py:121: UserWarning: Your `PyDataset` class should call `super().__init__(**kwargs)` in its constructor. `**kwargs` can include `workers`, `use_multiprocessing`, `max_queue_size`. Do not pass these arguments to `fit()`, as they will be ignored.
  self._warn_if_super_not_called()


6/6 - 19s - 3s/step - accuracy: 0.3475 - loss: 19.9331 - val_accuracy: 0.3702 - val_loss: 1.8577
Epoch 2/10
6/6 - 9s - 2s/step - accuracy: 0.3819 - loss: 2.0321 - val_accuracy: 0.2873 - val_loss: 1.4063
Epoch 3/10
6/6 - 9s - 1s/step - accuracy: 0.3434 - loss: 1.3543 - val_accuracy: 0.3646 - val_loss: 1.2459
Epoch 4/10
6/6 - 7s - 1s/step - accuracy: 0.3571 - loss: 1.2610 - val_accuracy: 0.4033 - val_loss: 1.2174
Epoch 5/10
6/6 - 8s - 1s/step - accuracy: 0.3874 - loss: 1.1629 - val_accuracy: 0.3702 - val_loss: 1.1315
Epoch 6/10
6/6 - 7s - 1s/step - accuracy: 0.3997 - loss: 1.1146 - val_accuracy: 0.4088 - val_loss: 1.0979
Epoch 7/10
6/6 - 8s - 1s/step - accuracy: 0.3887 - loss: 1.1327 - val_accuracy: 0.4088 - val_loss: 1.1082
Epoch 8/10
6/6 - 10s - 2s/step - accuracy: 0.3530 - loss: 1.1355 - val_accuracy: 0.4144 - val_loss: 1.1343
Epoch 9/10
6/6 - 9s - 2s/step - accuracy: 0.3324 - loss: 1.1509 - val_accuracy: 0.4033 - val_loss: 1.1472
Epoch 10/10
6/6 - 10s - 2s/step - accuracy: 0.4176 - l

# **Optimizacion**

In [ ]:
#Se aumento la cantidad de epocas para ver si mejora el modelo
n_epochs = 30
history = model.fit(train_generator, epochs=n_epochs, verbose=2, validation_data=validation_generator )

Epoch 1/30
6/6 - 10s - 2s/step - accuracy: 0.3489 - loss: 1.1402 - val_accuracy: 0.4033 - val_loss: 1.1505
Epoch 2/30
6/6 - 18s - 3s/step - accuracy: 0.3997 - loss: 1.1337 - val_accuracy: 0.2928 - val_loss: 1.1440
Epoch 3/30
6/6 - 11s - 2s/step - accuracy: 0.3695 - loss: 1.1188 - val_accuracy: 0.4309 - val_loss: 1.0983
Epoch 4/30
6/6 - 9s - 1s/step - accuracy: 0.4066 - loss: 1.1122 - val_accuracy: 0.4199 - val_loss: 1.0882
Epoch 5/30
6/6 - 9s - 2s/step - accuracy: 0.3929 - loss: 1.0971 - val_accuracy: 0.4088 - val_loss: 1.0839
Epoch 6/30
6/6 - 8s - 1s/step - accuracy: 0.4038 - loss: 1.0967 - val_accuracy: 0.4475 - val_loss: 1.0731
Epoch 7/30
6/6 - 9s - 1s/step - accuracy: 0.3970 - loss: 1.0926 - val_accuracy: 0.4033 - val_loss: 1.1196
Epoch 8/30
6/6 - 10s - 2s/step - accuracy: 0.4080 - loss: 1.0909 - val_accuracy: 0.3978 - val_loss: 1.1028
Epoch 9/30
6/6 - 8s - 1s/step - accuracy: 0.3901 - loss: 1.0959 - val_accuracy: 0.3923 - val_loss: 1.0990
Epoch 10/30
6/6 - 10s - 2s/step - accuracy

In [ ]:
#Se reduce el lote para ver si mejora el modelo

import os
import tensorflow as tf
from google.colab import drive

#Abrir Google Drive
drive.mount('/content/drive')

#Tamaño de lote
batch_size = 64

#Ruta a la carpeta de imagenes en Google Drive
carpeta_destino_base = '/content/drive/My Drive/BMBL_Clasificadas/'

#Crear un ImageDataGenerator para preprocesar las imagenes
train_datagen = tf.keras.preprocessing.image.ImageDataGenerator(
    zoom_range=1.0,
    validation_split=0.2  #Se agrega el porcentaje de validacion
)

#Crear un generador de entrenamiento desde las carpetas
train_generator = train_datagen.flow_from_directory(
    carpeta_destino_base,  #Directorio donde estan las imágenes
    target_size=(100, 100),  #Redimensionar las imagenes a 100x100
    #color_mode='grayscale',
    batch_size=batch_size,
    classes=['Comprar', 'Vender', 'Hold'],  #Clases
    subset='training'  #Entrenamiento
)

#Crear un generador de validacion
validation_generator = train_datagen.flow_from_directory(
    carpeta_destino_base,
    target_size=(100, 100),
    #color_mode='grayscale',
    batch_size=batch_size,
    classes=['Comprar', 'Vender', 'Hold'],
    subset='validation'  #Validacion
)

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
Found 728 images belonging to 3 classes.
Found 181 images belonging to 3 classes.


In [ ]:
#Se prueba con 30 epocas

model = tf.keras.models.Sequential([

    # Entrada
    tf.keras.layers.Input(shape=(100, 100, 3)), # 3 en color, 1 es grayscale

    # Convolución 1
    tf.keras.layers.Conv2D(16, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 2
    tf.keras.layers.Conv2D(32, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 3
    tf.keras.layers.Conv2D(64, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 4
    tf.keras.layers.Conv2D(64, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 5
    tf.keras.layers.Conv2D(64, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Achatamiento
    tf.keras.layers.Flatten(),

    # Capas densas
    tf.keras.layers.Dense(100, activation='relu'),

    # cantidad de neuronas de salida, una por clase (como son 3 clases, se coloca 3)
    tf.keras.layers.Dense(3, activation='softmax')

])

model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])

#model.summary()

n_epochs = 30
history = model.fit(train_generator, epochs=n_epochs, verbose=2, validation_data=validation_generator )

Epoch 1/30
12/12 - 20s - 2s/step - accuracy: 0.3352 - loss: 5.1158 - val_accuracy: 0.3757 - val_loss: 1.2072
Epoch 2/30
12/12 - 11s - 899ms/step - accuracy: 0.3448 - loss: 1.1478 - val_accuracy: 0.4088 - val_loss: 1.1013
Epoch 3/30
12/12 - 9s - 788ms/step - accuracy: 0.4066 - loss: 1.0941 - val_accuracy: 0.4144 - val_loss: 1.1149
Epoch 4/30
12/12 - 9s - 780ms/step - accuracy: 0.3668 - loss: 1.1086 - val_accuracy: 0.3536 - val_loss: 1.1173
Epoch 5/30
12/12 - 9s - 757ms/step - accuracy: 0.3984 - loss: 1.0983 - val_accuracy: 0.3923 - val_loss: 1.0966
Epoch 6/30
12/12 - 7s - 599ms/step - accuracy: 0.3970 - loss: 1.0879 - val_accuracy: 0.4033 - val_loss: 1.1054
Epoch 7/30
12/12 - 9s - 777ms/step - accuracy: 0.4176 - loss: 1.0857 - val_accuracy: 0.3204 - val_loss: 1.1069
Epoch 8/30
12/12 - 8s - 696ms/step - accuracy: 0.3420 - loss: 1.1171 - val_accuracy: 0.3204 - val_loss: 1.1112
Epoch 9/30
12/12 - 9s - 758ms/step - accuracy: 0.4190 - loss: 1.0923 - val_accuracy: 0.4144 - val_loss: 1.1055
Ep

In [ ]:
#Se reduce nuevamente el lote para ver si mejora el modelo

import os
import tensorflow as tf
from google.colab import drive

#Abrir Google Drive
drive.mount('/content/drive')

#Tamaño de lote
batch_size = 16

#Ruta a la carpeta de imagenes en Google Drive
carpeta_destino_base = '/content/drive/My Drive/BMBL_Clasificadas/'

#Crear un ImageDataGenerator para preprocesar las imagenes
train_datagen = tf.keras.preprocessing.image.ImageDataGenerator(
    zoom_range=1.0,
    validation_split=0.2  #Se agrega el porcentaje de validacion
)

#Crear un generador de entrenamiento desde las carpetas
train_generator = train_datagen.flow_from_directory(
    carpeta_destino_base,  #Directorio donde estan las imágenes
    target_size=(100, 100),  #Redimensionar las imagenes a 100x100
    #color_mode='grayscale',
    batch_size=batch_size,
    classes=['Comprar', 'Vender', 'Hold'],  #Clases
    subset='training'  #Entrenamiento
)

#Crear un generador de validacion
validation_generator = train_datagen.flow_from_directory(
    carpeta_destino_base,
    target_size=(100, 100),
    #color_mode='grayscale',
    batch_size=batch_size,
    classes=['Comprar', 'Vender', 'Hold'],
    subset='validation'  #Validacion
)

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).
Found 728 images belonging to 3 classes.
Found 181 images belonging to 3 classes.


In [ ]:
#Se prueba con 30 epocas

model = tf.keras.models.Sequential([

    # Entrada
    tf.keras.layers.Input(shape=(100, 100, 3)), # 3 en color, 1 es grayscale

    # Convolución 1
    tf.keras.layers.Conv2D(16, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 2
    tf.keras.layers.Conv2D(32, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 3
    tf.keras.layers.Conv2D(64, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 4
    tf.keras.layers.Conv2D(64, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Convolución 5
    tf.keras.layers.Conv2D(64, (3,3), activation='relu'),
    tf.keras.layers.MaxPooling2D(2,2),

    # Achatamiento
    tf.keras.layers.Flatten(),

    # Capas densas
    tf.keras.layers.Dense(100, activation='relu'),

    # cantidad de neuronas de salida, una por clase (como son 3 clases, se coloca 3)
    tf.keras.layers.Dense(3, activation='softmax')

])

model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])

#model.summary()

n_epochs = 30
history = model.fit(train_generator, epochs=n_epochs, verbose=2, validation_data=validation_generator )

Epoch 1/30
46/46 - 17s - 362ms/step - accuracy: 0.3915 - loss: 1.9147 - val_accuracy: 0.3481 - val_loss: 1.1330
Epoch 2/30
46/46 - 13s - 279ms/step - accuracy: 0.3860 - loss: 1.1191 - val_accuracy: 0.4088 - val_loss: 1.0987
Epoch 3/30
46/46 - 11s - 234ms/step - accuracy: 0.3901 - loss: 1.1122 - val_accuracy: 0.4088 - val_loss: 1.1013
Epoch 4/30
46/46 - 11s - 231ms/step - accuracy: 0.4080 - loss: 1.0870 - val_accuracy: 0.4033 - val_loss: 1.0853
Epoch 5/30
46/46 - 7s - 156ms/step - accuracy: 0.3887 - loss: 1.0939 - val_accuracy: 0.3978 - val_loss: 1.0916
Epoch 6/30
46/46 - 10s - 224ms/step - accuracy: 0.3997 - loss: 1.0863 - val_accuracy: 0.3204 - val_loss: 1.1081
Epoch 7/30
46/46 - 11s - 234ms/step - accuracy: 0.4025 - loss: 1.0853 - val_accuracy: 0.3260 - val_loss: 1.1613
Epoch 8/30
46/46 - 8s - 183ms/step - accuracy: 0.3942 - loss: 1.0977 - val_accuracy: 0.3978 - val_loss: 1.0825
Epoch 9/30
46/46 - 7s - 162ms/step - accuracy: 0.4093 - loss: 1.0874 - val_accuracy: 0.4144 - val_loss: 1.

Finalmente se concluye que el modelo con lote 64 y 30 epocas es el que obtuvo mejores métricas, ya que alcanzó la mejor precisión de validación (44.20%) y también tuvo una pérdida de validación relativamente baja (1.0775), lo que sugiere un mejor rendimiento general.

# **Informe**

https://docs.google.com/document/d/1wJQSY3L7aoGhmHmMvnr3kl5YWY0-e2c6S4KOkAXbD5Y/edit?usp=sharing